openassessment/xblock/openassessmentblock.py

```diff
@@ -352,6 +352,34 @@ def get_anonymous_user_id(self, username, course_id):
         """
         return self.runtime.service(self, 'user').get_anonymous_user_id(username, course_id)
 
+    def is_user_state_service_available(self):
+        """
+        Check if the user state service is present in runtime.
+        """
+        return self.runtime._services.get('user_state') is not None  # pylint: disable=protected-access
```

Comment:
I think if you decorate the `OpenAssessmentBlock` class with `@XBlock.needs('user_state')`, you'll be able to call `self.runtime.service(self, 'user_state')` without having to look at the private `_services` attribute.

Reply:
Sure.